traffic_ops/install/bin/postinstall.py

```diff
@@ -560,48 +703,52 @@ def unmarshal_config(dct: dict) -> Dict[str, List[Question]]:
 
 	return ret
 
-def setup_maxmind(maxmind_answer: str, root: str):
+def setup_maxmind(maxmind_answer, root): # type: (str, str) -> None
 	"""
 	If 'maxmind_answer' is a truthy response ('y' or 'yes' (case-insensitive), sets up a Maxmind
 	database using `wget`.
 	"""
-	if maxmind_answer.casefold() not in {'y', 'yes'}:
+	if maxmind_answer.lower() not in {'y', 'yes'}:
 		logging.info("Not downloading Maxmind data")
 		return
 
 	os.chdir(os.path.join(root, 'opt/traffic_ops/app/public/routing'))
 
+	def failed_download(e, ip_version):  # type: (Exception, int) -> None
+		logging.error("Failed to download MaxMind data")
+		logging.debug("(ipv%d) Exception: %s", ip_version, e)
+
 	wget = "/usr/bin/wget"
 	cmd = [wget, "https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz"]
 	# Perl ignored errors downloading the databases, so we do too
 	try:
-		subprocess.run(
+		subprocess.check_call(
 			cmd,
 			stderr=subprocess.PIPE,
 			stdout=subprocess.PIPE,
-			check=True,
 			universal_newlines=True
 		)
+	except subprocess.CalledProcessError as e:
+		failed_download(e, 4)
 	except subprocess.SubprocessError as e:
-		logging.error("Failed to download MaxMind data")
-		logging.debug("(ipv4) Exception: %s", e)
+		failed_download(e, 4)
 
 	cmd[1] = (
 		"https://geolite.maxmind.com/download/geoip/database/GeoLiteCityv6-beta/GeoLiteCityv6.dat.gz"
 	)
 	try:
-		subprocess.run(
+		subprocess.check_call(
 			cmd,
 			stderr=subprocess.PIPE,
 			stdout=subprocess.PIPE,
-			check=True,
 			universal_newlines=True
 		)
+	except subprocess.CalledProcessError as e:
+		failed_download(6)
```

Comment:
passing `6` as `Exception` - missing those type hints already? :P

Reply:
PyCharm gave me the warnings, I just didn't scroll down.
Passing 6 as an `int` in ab6881f0fb